src/Controls/src/Core/Handlers/Items2/iOS/StructuredItemsViewController2.cs

```diff
@@ -36,194 +40,147 @@ protected override void Dispose(bool disposing)
 
 			if (disposing)
 			{
-				if (_headerViewFormsElement != null)
-				{
-					//_headerViewFormsElement.MeasureInvalidated -= OnFormsElementMeasureInvalidated;
-				}
-
-				if (_footerViewFormsElement != null)
-				{
-					//_footerViewFormsElement.MeasureInvalidated -= OnFormsElementMeasureInvalidated;
-				}
 
-				_headerUIView = null;
-				_headerViewFormsElement = null;
-				_footerUIView = null;
-				_footerViewFormsElement = null;
 			}
 
 			base.Dispose(disposing);
 		}
 
-		protected override bool IsHorizontal => (ItemsView?.ItemsLayout as ItemsLayout)?.Orientation == ItemsLayoutOrientation.Horizontal;
-
-		protected override CGRect DetermineEmptyViewFrame()
-		{
-			nfloat headerHeight = 0;
-			var headerView = CollectionView.ViewWithTag(HeaderTag);
-
-			if (headerView != null)
-				headerHeight = headerView.Frame.Height;
-
-			nfloat footerHeight = 0;
-			var footerView = CollectionView.ViewWithTag(FooterTag);
-
-			if (footerView != null)
-				footerHeight = footerView.Frame.Height;
-
-			return new CGRect(CollectionView.Frame.X, CollectionView.Frame.Y, CollectionView.Frame.Width,
-				Math.Abs(CollectionView.Frame.Height - (headerHeight + footerHeight)));
-		}
-
-		public override void ViewWillLayoutSubviews()
+		protected virtual string DetermineViewReuseId(NSString elementKind)
```

Comment:
I m not sure about this implementation. We maybe should just  flat this to ItemsViewController